### ISSUE TITLE
🔬 Retrain with BioMistral-7B-SLERP on Cleaned Dataset

### ISSUE THREAD

**Issue by albertoclemente** [albertoclemente/medical-ner-fine-tuning#2]
## 🎯 Objective

Retrain the medical NER model using **BioMistral-7B-SLERP** instead of Llama-3.2-3B-Instruct, leveraging the newly cleaned and optimized dataset.

---

## 📊 Dataset Ready

The cleaned dataset is available in `data/splits_cleaned_20251113/`:
- **Train**: 2,397 samples (799/798/800 per task)
- **Validation**: 298 samples (99/99/100 per task)
- **Test**: 299 samples (100/99/100 per task)

**Quality**: 99.8% retention, all samples validated, perfect stratification

---

## 🔧 Model Configuration

**New Base Model**: `BioMistral/BioMistral-7B-SLERP`

**Why BioMistral-7B-SLERP?**
- SLERP merge of medical domain models
- Enhanced medical/biomedical knowledge
- Larger capacity (7B vs 3B parameters)
- Superior performance on medical NER tasks
- Improved entity disambiguation

---

### 1. Update Training Notebook
File: `Medical_NER_Fine_Tuning_RUN.ipynb` (or create new version)

**Changes needed**:
```python
# OLD
MODEL_NAME = "meta-llama/Llama-3.2-3B-Instruct"

# NEW
MODEL_NAME = "BioMistral/BioMistral-7B-SLERP"

# Dataset paths
TRAIN_DATA = "data/splits_cleaned_20251113/train.jsonl"
VAL_DATA = "data/splits_cleaned_20251113/validation.jsonl"
```

### 2. System Prompt Adjustment
Keep medical NER-specific prompt but adapt if needed for BioMistral format

### 3. Training Hyperparameters
Consider adjusting for 7B SLERP model:
- Learning rate: May need tuning (start with 2e-5)
- Batch size: Adjust based on GPU memory
- LoRA rank: Consider r=16 or r=32 for larger model
- Gradient accumulation: Increase if memory constrained

---

## ✅ Expected Improvements

- Superior medical entity recognition (SLERP-merged domain knowledge)
- Improved chemical/disease disambiguation
- More accurate relationship extraction
- Higher overall F1 scores (target: 70-80%)
- Better handling of complex medical terminology

---

## 🚀 Action Items

- [ ] Update training notebook with BioMistral-7B-SLERP
- [ ] Configure LoRA parameters for 7B model
- [ ] Update dataset paths to `splits_cleaned_20251113/`
- [ ] Run training on RunPod with A100 GPU
- [ ] Monitor training metrics on W&B
- [ ] Save checkpoints to HuggingFace Hub
- [ ] Evaluate on test set using fixed evaluation notebook
- [ ] Compare results with Llama-3.2-3B baseline

---

## 📚 References

- BioMistral-7B-SLERP: https://huggingface.co/BioMistral/BioMistral-7B-SLERP
- Cleaned dataset: `data/splits_cleaned_20251113/`
- Evaluation notebook: `notebooks/evaluation/Medical_NER_Evaluation_run04_20251113.ipynb`
- Optimization docs: `docs/OPTIMIZATION_SUMMARY.md`
